src/validation/validation.cpp

```diff
@@ -2809,17 +2802,8 @@ void UpdateTip(CBlockIndex *pindexNew)
         CheckAndAlertUnknownVersionbits(chainParams, chainActive.Tip());
     }
 
-    if (IsNov152018Scheduled()) // Set the global variables based on the fork state of the NEXT block
-    {
-        if (IsNov152018Enabled(chainParams.GetConsensus(), pindexNew))
-        {
-            enableCanonicalTxOrder = true;
-        }
-        else
-        {
-            enableCanonicalTxOrder = false;
-        }
-    }
+    // Set the global variables based on the fork state of the NEXT block
+    enableCanonicalTxOrder = IsNov152018Scheduled() && IsNov152018Enabled(chainParams.GetConsensus(), pindexNew);
```

Comment:
If I am not mistaken, this latter part is a subtle logic change. `enableCanonicalTxOrder` is not set unless the `IsNov152018Scheduled()` flag is true and if I see it correctly, this has also been caught by the test suite (albeit in a pretty convoluted way).

Otherwise, good catch!

Reply:
hmm, not sure if the test suite failed due to a failed test or due to time outs or library install errors, it passed on some architectures.

Unless `enableCanonicalTxOrder` was set to `true` before the original `if` I don't see why it'd be a subtle logic change. My thinking was that you don't want to have ordering enabled a priori and that it cannot happen unless those two conditions are met.

Reply:
@gubatron: Good points. I thought this was meant to be this way to be able use the tweak unless it is overriden. But I might be wrong, though. @gandrewstone knows better :)